test/integration/segmentation_camera.cc

```diff
@@ -0,0 +1,262 @@
+void SegmentationCameraTest::SegmentationCameraBoxes(
+  const std::string &_renderEngine)
+{
+  // Optix is not supported
+  if (_renderEngine.compare("optix") == 0)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' doesn't support depth cameras" << std::endl;
+    return;
+  }
+
+  // Setup ign-rendering with an empty scene
+  auto *engine = ignition::rendering::engine(_renderEngine);
+  if (!engine)
+  {
+    igndbg << "Engine '" << _renderEngine
+              << "' is not supported" << std::endl;
+    return;
+  }
+
+  ignition::rendering::ScenePtr scene = engine->CreateScene("scene");
+  BuildScene(scene);
+
+  // Create Segmentation camera
+  auto camera = scene->CreateSegmentationCamera("SegmentationCamera");
+  ASSERT_NE(camera, nullptr);
+
+  camera->SetLocalPosition(0.0, 0.0, 0.0);
+  camera->SetLocalRotation(0.0, 0.0, 0.0);
+
+  int backgroundLabel = 23;
+  camera->SetBackgroundLabel(backgroundLabel);
+  EXPECT_EQ(camera->BackgroundLabel(), backgroundLabel);
+
+  camera->SetSegmentationType(SegmentationType::ST_SEMANTIC);
+
+  EXPECT_EQ(camera->Type(), SegmentationType::ST_SEMANTIC);
+
+  camera->EnableColoredMap(false);
+  EXPECT_FALSE(camera->IsColoredMap());
+
+  int width = 320;
+  int height = 240;
+  double aspectRatio = width/height;
+
+  camera->SetAspectRatio(aspectRatio);
+  camera->SetImageWidth(width);
+  camera->SetImageHeight(height);
+  camera->SetHFOV(IGN_PI / 2);
+
+  // add camera to the scene
+  scene->RootVisual()->AddChild(camera);
+
+  // Set a callback on the  camera sensor to get a Segmentation camera frame
+  ignition::common::ConnectionPtr connection =
+      camera->ConnectNewSegmentationFrame(
+          std::bind(OnNewSegmentationFrame,
+          std::placeholders::_1, std::placeholders::_2, std::placeholders::_3,
+          std::placeholders::_4, std::placeholders::_5));
+  EXPECT_NE(nullptr, connection);
+
+  // Update once to create image
+  camera->Update();
+
+  // Semantic Test
+
+  // get the center of each box, the percentages locates the center
+  math::Vector2d leftProj(width * 0.25, height * 0.5);
+  math::Vector2d rightProj(width * 0.75, height * 0.5);
+  math::Vector2d middleProj(width * 0.5, height * 0.5);
+
+  // get their index in the buffer
+  uint32_t leftIndex = (leftProj.Y() * width + leftProj.X()) * 3;
+  uint32_t rightIndex = (rightProj.Y() * width + rightProj.X()) * 3;
+  uint32_t middleIndex = (middleProj.Y() * width + middleProj.X()) * 3;
+
+  // test
+  g_mutex.lock();
+  g_counter = 0;
+
+  // cast the unsigned char to unsigned int to read it
+  uint8_t leftLabel =   g_buffer[leftIndex];
+  uint8_t rightLabel =  g_buffer[rightIndex];
+  uint8_t middleLabel = g_buffer[middleIndex];
+
+  // check the label
+  EXPECT_EQ(leftLabel  , 1);
+  EXPECT_EQ(middleLabel, 2);
+  EXPECT_EQ(rightLabel , 1);
+
+  // check if the first pixel(background) = the background label
+  int background = g_buffer[0];
+  EXPECT_EQ(background, backgroundLabel);
+
+  g_mutex.unlock();
```

Comment:
```suggestion
  // cast the unsigned char to unsigned int to read it
  uint8_t leftLabel =   g_buffer[leftIndex];
  uint8_t rightLabel =  g_buffer[rightIndex];
  uint8_t middleLabel = g_buffer[middleIndex];

  // check the label
  EXPECT_EQ(leftLabel  , 1);
  EXPECT_EQ(middleLabel, 2);
  EXPECT_EQ(rightLabel , 1);

  // check if the first pixel(background) = the background label
  int background = g_buffer[0];
  EXPECT_EQ(background, backgroundLabel);
```
I don't think you need to lock/unlock the mutex since you're not calling `camera->Update`